src/main/java/pl/cleankod/exchange/entrypoint/AccountController.java

```diff
@@ -1,57 +1,68 @@
-import pl.cleankod.exchange.core.usecase.FindAccountAndConvertCurrencyUseCase;
-import pl.cleankod.exchange.core.usecase.FindAccountUseCase;
+import pl.cleankod.exchange.provider.FindAccountService;
 
 import java.net.URLDecoder;
 import java.nio.charset.StandardCharsets;
-import java.util.Currency;
-import java.util.Optional;
 
 @RestController
 @RequestMapping("/accounts")
+@OpenAPIDefinition(info = @Info(title = "Accounts API", version = "1.0", description = "API for accessing to the customer account"))
+@ApiResponses({
+        @ApiResponse(description = "Successfully returned account object", responseCode = "200"),
+        @ApiResponse(
+                description = "Client sent an invalid request",
+                responseCode = "400",
+                content = @Content(examples = @ExampleObject(value = "{\"message\":\"Client sent an invalid request\"}"))
+        ),
+        @ApiResponse(
+                description = "Server failed to fulfill a valid request due to an error with server",
+                responseCode = "500",
+                content = @Content(examples = @ExampleObject(value = "{\"message\":\"Server failed to fulfill a valid request due to an error with server\"}"))
+        )
+})
 public class AccountController {
 
-    private final FindAccountAndConvertCurrencyUseCase findAccountAndConvertCurrencyUseCase;
-    private final FindAccountUseCase findAccountUseCase;
+    private final FindAccountService findAccountService;
 
-    public AccountController(FindAccountAndConvertCurrencyUseCase findAccountAndConvertCurrencyUseCase,
-                             FindAccountUseCase findAccountUseCase) {
-        this.findAccountAndConvertCurrencyUseCase = findAccountAndConvertCurrencyUseCase;
-        this.findAccountUseCase = findAccountUseCase;
+    public AccountController(FindAccountService findAccountService) {
+        this.findAccountService = findAccountService;
     }
 
     @GetMapping(path = "/{id}")
+    @Parameters({
+            @Parameter(name = "id", description = "Account id"),
+            @Parameter(name = "currency", description = "Account currency")
+    })
     public ResponseEntity<Account> findAccountById(@PathVariable String id, @RequestParam(required = false) String currency) {
-        return Optional.ofNullable(currency)
-                .map(s ->
-                        findAccountAndConvertCurrencyUseCase.execute(Account.Id.of(id), Currency.getInstance(s))
-                                .map(ResponseEntity::ok)
-                                .orElse(ResponseEntity.notFound().build())
-                )
-                .orElseGet(() ->
-                        findAccountUseCase.execute(Account.Id.of(id))
-                                .map(ResponseEntity::ok)
-                                .orElse(ResponseEntity.notFound().build())
+        return findAccountService.findAccountById(Account.Id.of(URLDecoder.decode(id, StandardCharsets.UTF_8)), currency)
```

Comment:
The presentation layer should know nothing about the domain. According to clean architecture, onion architecture, layered architecture, and many others, the dependencies direction is always downwards. IMO the application layer should give you ready to use DTO with the data you need.